crates/walrus-sui/src/client/retry_client.rs

```diff
@@ -1409,18 +1444,33 @@ impl RetriableRpcClient {
         &self,
         sequence_number: u64,
     ) -> Result<CheckpointData, RetriableClientError> {
+        let start_time = Instant::now();
         let error = match self.get_full_checkpoint_from_primary(sequence_number).await {
-            Ok(checkpoint) => return Ok(checkpoint),
+            Ok(checkpoint) => {
+                self.record_rpc_latency("get_full_checkpoint", "success", start_time.elapsed());
+                return Ok(checkpoint);
+            }
             Err(error) => error,
         };
 
         tracing::debug!(?error, "primary client error while fetching checkpoint");
         let Some(ref fallback) = self.fallback_client else {
+            self.record_rpc_latency("get_full_checkpoint", "failure", start_time.elapsed());
             return Err(error);
         };
 
-        self.get_full_checkpoint_from_fallback_with_retries(fallback, sequence_number)
-            .await
+        let fallback_start_time = Instant::now();
+        let result = self
+            .get_full_checkpoint_from_fallback_with_retries(fallback, sequence_number)
+            .await;
+
+        self.record_fallback_metrics(
+            "get_full_checkpoint",
+            &result,
+            fallback_start_time.elapsed(),
+        );
+
```

Comment:
Suggestion: may also be interesting to record the whole function latency from `start_time` until it returns. Maybe use `monitored_scope`, which records the time spent in this whole function until return?